packages/create-modular-react-app/src/index.ts

```diff
@@ -1,34 +1,157 @@
+  name: string;
+  repo?: boolean;
+  preferOffline?: boolean;
+}): Promise<void> {
+  const preferOfflineArg = argv.preferOffline ? ['--prefer-offline'] : [];
+  if (isYarnInstalled() === false) {
+    console.error(
+      'Please install `yarn` before attempting to run `create-modular-react-app`.',
+    );
+    process.exit(1);
+  }
+
+  const name = argv.name;
+  const newModularRoot =
+    name[0] === '/' || name.includes(':\\')
+      ? /* absolute */ name
+      : path.join(process.cwd(), name);
+  const projectPackageJsonPath = path.join(newModularRoot, 'package.json');
+  const templatePath = path.join(__dirname, '..', 'template');
+
+  // Create a new CRA app, modify config for workspaces
+  fs.mkdirpSync(newModularRoot);
+
+  // CRA bails from creating a Git repository if it's run within one.
+  //
+  // See: https://github.com/facebook/create-react-app/blob/47e9e2c7a07bfe60b52011cf71de5ca33bdeb6e3/packages/react-scripts/scripts/init.js#L48-L50
+  if (argv.repo !== false) {
+    execSync('git', ['init'], {
+      cwd: newModularRoot,
+    });
+  }
+
+  execSync('yarnpkg', ['init', '-y'], {
+    cwd: newModularRoot,
   });
-});
 
-try {
-  void program.parseAsync(process.argv.slice(2));
-} catch (err) {
-  console.error(err);
+  fs.writeJsonSync(projectPackageJsonPath, {
+    ...fs.readJsonSync(projectPackageJsonPath),
+    private: true,
+    workspaces: ['packages/**'],
+    modular: {
+      type: 'root',
+    },
+    scripts: {
+      start: 'modular start app',
+      build: 'modular build app',
+      test: 'modular test',
+      lint: 'eslint . --ext .js,.ts,.tsx',
+      prettier: 'prettier --write .',
+    },
+    eslintConfig: {
+      extends: 'modular-app',
+    },
+    prettier: {
+      singleQuote: true,
+      trailingComma: 'all',
+      printWidth: 80,
+      proseWrap: 'always',
+    },
+  });
+
+  execSync(
+    'yarnpkg',
+    [
+      'add',
+      '-W',
+      ...preferOfflineArg,
+      '@testing-library/dom',
+      '@testing-library/jest-dom',
+      '@testing-library/react',
+      '@testing-library/user-event',
+      '@types/node',
+      '@types/jest',
+      '@types/react',
+      '@types/react-dom',
+      'react',
+      'react-dom',
+      'prettier',
+      'modular-scripts',
+      'eslint',
+      'eslint-config-modular-app',
+      'typescript@^4.1.2',
+    ],
+    { cwd: newModularRoot },
+  );
+
+  fs.copySync(templatePath, newModularRoot);
+
+  // rename gitgnore to .gitgnore so it actually works
+  fs.moveSync(
+    path.join(newModularRoot, 'gitignore'),
+    path.join(newModularRoot, '.gitignore'),
+  );
+
+  execSync(
+    'yarnpkg',
+    [
+      'modular',
+      'add',
+      'app',
+      '--unstable-type',
+      'app',
+      '--unstable-name',
+      'app',
+    ],
+    {
+      cwd: newModularRoot,
+    },
+  );
+
+  if (argv.repo !== false) {
+    execSync('git', ['add', '.'], {
+      cwd: newModularRoot,
+    });
+
+    // don't try to commit in CI
+    if (!process.env.CI) {
```

Comment:
Checking for CI tests in the source code has a code smell... And I don't know if we should be committing automatically for projects. This will violate the webhook that checks for JIRA tickets in the commits.

Reply:
I'm happy to remove the `CI` check - I'm not sure why this would be being used in CI anyway.